libkodimote/kodi.cpp

```diff
@@ -498,14 +512,28 @@ void Kodi::setVolume(int volume)
 
     } else {
         if(volume != m_volume) {
-            QVariantMap map;
-            map.insert("volume", volume);
-            KodiConnection::sendCommand("Application.SetVolume", map);
-
+            requestVolume(volume);
         }
     }
 }
 
+void Kodi::requestVolume(int volume) {
+    m_volumePending = volume;
+    if (!m_volumeChangeOutstanding) {
+        m_volumeChangeOutstanding = true;
+        QVariantMap map;
+        map.insert("volume", volume);
+        KodiConnection::sendCommand("Application.SetVolume", map);
```

Comment:
Hi, thanks. This is much better already. I think we can get rid of the m_volumeTimer workaround now. All you'd have to do is to add two parameters here like this:

KodiConnection::sendCommand("Application.SetVolume", map, this, "volumeReceived");

and in volumeReceived() set the m_volimeChangeOutstanding to false. In my experience replies to commands come reliably and it shouldn't require to fallback on a timer.